lib/features/thread/data/repository/thread_repository_impl.dart

```diff
@@ -348,4 +280,59 @@ class ThreadRepositoryImpl extends ThreadRepository {
       },
     );
   }
+
+  Future<void> _synchronizeCacheWithChanges(
+    AccountId accountId,
+    State currentState,
+    {
+      Properties? propertiesCreated,
+      Properties? propertiesUpdated,
+    }
+  ) async {
+    final localEmailList = await mapDataSource[DataSourceType.local]!
+        .getAllEmailCache();
```

Comment:
IMO, if you do that you will have to get all the data in the email table out. Is that necessary? Does it take too long to query? Why don't you add `filter` conditions.

Reply:
yes, it will take time to query. But if not, it will be wrong in the cache, some messages never be updated after state update (some issues you can see before).

Reply:
this list will be used to `combine` between cache and `network response` (cache hit or not) to update cache. So, add `filter` here will make the cache update wrong